src/JustEat.StatsD/PooledUdpTransport.cs

```diff
@@ -10,13 +11,71 @@ namespace JustEat.StatsD
     /// </summary>
     public sealed class PooledUdpTransport : IStatsDTransport, IDisposable
     {
-        private readonly SimpleObjectPool<Socket> _socketPool
-            = new SimpleObjectPool<Socket>(
-                Environment.ProcessorCount,
-                pool => UdpTransport.CreateSocket());
+        private sealed class ConnectedPool : IDisposable
+        {
+            private bool _disposed;
+
+            public ConnectedPool(IPEndPoint ipEndPoint)
+            {
+                IpEndPoint = ipEndPoint;
+
+                Pool = new SimpleObjectPool<Socket>(
+                    Environment.ProcessorCount,
+                    pool =>
+                    {
+                        var socket = UdpTransport.CreateSocket();
+                        socket.Connect(ipEndPoint);
```

Comment:
Need to `try-catch` around `Connect()` and dispose in the catch block, as otherwise if it fails we'll leak a socket.